.github/workflows/publish.yml

```diff
@@ -0,0 +1,40 @@
+name: publish
+
+on:
+  workflow_call:
+    inputs: 
+      source:
+        required: true
+        type: string
+    secrets:
+      apikey:
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get the version
+        id: get_version
+        run: |
+          VERSION="${GITHUB_REF_NAME//v}"
+          echo VERSION:${VERSION}
+          echo ::set-output name=VERSION::${VERSION}  
+      - uses: actions/checkout@v2    
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+      - name: Install dependencies
+        run: dotnet restore -p:NuGetBuild=True
+      - name: Build
+        run: dotnet build --configuration Release --no-restore -p:NuGetBuild=True
+      - name: Test
+        run: dotnet test --no-restore -p:NuGetBuild=True
```

Comment:
To share the build pipeline it needs to upload and download the artifacts produced. So takes longer, and is harder to configure. Easier just to have two files - not that much to maintain I think

Reply:
Why do we need a separate build pipeline at all though?

Reply:
Well that’s up to you i guess. 
I like it because it builds on pushes, so you know before you try to release that it actually builds ok. But can remove it if you don’t want that behavior?

Reply:
It could make some sense for custom branches and PRs but for the dev branch it's too late, so we don't need it. (I don't really see why we'd need it anyway though since the publish pipeline will fail anyway if the build has an issue.)

Reply:
It’s useful if you do releases from the GitHub ui as that involves tags and notifications that would still be sent. 
but if you don’t want it, it’s no issue to drop it :) just let me know what you prefer

Reply:
I'm not sure I understand. Let me explain: We'll publish releases by pushing tags to `dev`, which will publish to NuGet too if all goes well, and also add releases on GitHub corresponding to the tag (while we didn't talk about this before, we'll indeed do this). What's in `dev` already passed code review and build/testing in the OSOCE repo.

In this approach, where would having a separate build pipeline for each submodule help? If we still commit something to the submodule's `dev` that's breaking (can happen), and would break this build, it'll break the NuGet publish build anyway, no? At that point we'll fix it, move the tag, wait for the NuGet publish to succeed, and only publish the release on GitHub after that.

Reply:
Sure that makes sense, obviously our workflows are different and I know less if yours. 
I do a lot of my releases through the GitHub ui, without ever opening code locally, so having a build prior to making a release is useful to me.
It maybe useful for you, as when you are updating versions nuget will take 5-10 minutes to index a new version, so when you update the HL version, updating HE will not build until nuget has been nieces it, so it can be convenient just to push all the versions updates to everything at nice and wait til the prs build successfully, for example.

But its just a matter of taste/preference, so if you don’t see the need for it, let me know and I’ll drop it out

Reply:
I see, I think what you do only from GitHub is not something I've done or heard about. Can you attach a short screen share so I can see what you mean?

Reply:
Sent you a screen share of a release through the GitHub ui.

The build step still optional. Let me know if you want it removed.

Reply:
I see, so you're adding the version tag when creating the release from GitHub as opposed to pushing the tag and creating a release targeting that. And you edited some package reference version on GitHub.

Let's leave the build pipeline too.